File: vol2/vol2-java-examples/src/main/java/com/heatonresearch/aifh/general/collections/ObjectHolder.java
package com.heatonresearch.aifh.general.collections;

import java.io.Serializable;

/**
 * Holds an object of the specified type.  Used with the ObjectChooser.
 * Associates an object with a probability.
 *
 * @param <T> The type of object to hold.
 */
public class ObjectHolder<T> implements Serializable {
    /**
     * The object to hold.
     */
    private final T obj;

    /**
     * The probability.
     */
    private final double probability;

    public ObjectHolder(T theObj, double probability) {
        this.obj = theObj;
        this.probability = probability;
    }

    /**
     * @return the opp
     */
    public T getObj() {
        return this.obj;
    }

    /**
     * @return the probability
     */
    public double getProbability() {
        return probability;
    }
}

File: vol2/vol2-csharp-examples/AIFH-Vol2-Core/Core/General/Collections/ObjectHolder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AIFH_Vol2.Core.General.Collections
{
    /// <summary>
    /// Holds an object of the specified type.  Used with the ObjectChooser.
    /// Associates an object with a probability.
    /// </summary>
    public class ObjectHolder<T>
    {
        /// <summary>
        /// The object to hold.
        /// </summary>
        private readonly T _obj;

        /// <summary>
        /// The probability.
        /// </summary>
        private double _probability;

        public ObjectHolder(T theObj, double probability)
        {
            _obj = theObj;
            _probability = probability;
        }

        /// <summary>
        /// The object.
        /// </summary>
        public T Obj
        {
            get
            {
                return _obj;
            }
        }

        /// <summary>
        /// The probability.
        /// </summary>
        public double Probability
        {
            get
            {
                return _probability;
            }
        }
    }
}
